packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -548,11 +559,14 @@ class ViewSettingsDialog extends UI5Element {
 		const _currentSortOrderSelected = this._currentSettings.sortOrder.filter(item => item.selected)[0],
 			_currentSortBySelected = this._currentSettings.sortBy.filter(item => item.selected)[0],
 			sortOrder = _currentSortOrderSelected && _currentSortOrderSelected.text,
-			sortBy = _currentSortBySelected && _currentSortBySelected.text;
-
+			sortDescending = !this._currentSettings.sortOrder[0].selected,
+			sortBy = _currentSortBySelected && _currentSortBySelected.text,
+			sortByItem = sortBy && this.initSortByItems.filter(item => item.text === sortBy)[0].associatedItem;
```

Comment:
Just to make sure I won't overlook something. What will happen if the app developer uses two sort items with the same name (although does not make sense).

Then would not this always return the first
this.initSortByItems.filter(item => item.text === sortBy)[0].associatedItem;

discovering the associated item by filtering the sort items by text does not seem robust.

Check it out and if it fails as described above, please create a separate issue, so that this is improved in future or you can try fixing it here. 

One way or another, you can see a solution pattern in the Shellbar.js/ShellBar.hbs where `refItemId` is set to the internally rendered items (ui5-li), this refId maps to the external element (in your case ui5-sort-item). Then to get the external element you can: 
```js
const sortByItem = this.initSortByItems.find(item => {
				return item._id === refItemId;
	});
```

Reply:
IMO using the same named Sort Items is not a good idea in general...